### ISSUE TITLE
New Rule: implement `markdownlint`'s `MD027`

### ISSUE THREAD

**Issue by tooth-is-silver** [eslint-markdown/eslint-markdown#744]
### Rule details

Add a rule equivalent to markdownlint's MD027.

The rule should report blockquote markers followed by more than one space and provide an autofix that removes the extra spaces.

Support a `listItems` option that controls whether ordered and unordered list items inside blockquotes are checked. Default: `true`.

### What type of rule is this?

Enforces a consistent style or user preference

### Example code

```markdown
<!-- Incorrect -->

>  This blockquote has too many spaces.
>  Only one space should follow the marker.

<!-- Correct -->

> This blockquote has one space.
> Its indentation is consistent.
```

### Prior Art

- markdownlint: [`MD027` - Multiple spaces after blockquote symbol](https://github.com/DavidAnson/markdownlint/blob/main/doc/md027.md)
